File: PizzeriaApi.Contracts/Shared/Pizza.cs
namespace PizzeriaApi.Contracts.Shared;

public record Pizza(
    int Id,
    string Name,
    string ImageUrl,
    int ImageId,
    string Description,
    bool IsActive,
    List<PizzaPrice> Prices,
    List<PizzaIngredient> Ingredients)
{
    public static Pizza CreateEmpty() => new(
        Id: 0,
        Name: string.Empty,
        ImageUrl: string.Empty,
        ImageId: 0,
        Description: string.Empty,
        IsActive: true,
        Prices: [],
        Ingredients: []);
}

File: Web/Management.Web/src/models/pizza/pizza.ts
import {PizzaPrice} from "@/models/pizza/pizzaPrice";
import {PizzaIngredient} from "@/models/pizza/pizzaIngredient";

export interface Pizza {
    id: number;
    name: string;
    imageUrl: string;
    imageId: number;
    description: string;
    isActive: boolean;
    prices: PizzaPrice[];
    ingredients: PizzaIngredient[]
}